core/trino-spi/src/main/java/io/trino/spi/ptf/TableArgument.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.ptf;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.trino.spi.expression.ConnectorExpression;
+import io.trino.spi.type.RowType;
+
+import java.util.List;
+import java.util.Optional;
+
+import static io.trino.spi.ptf.ConnectorTableFunction.checkNotNullOrEmpty;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * This class represents the table argument passed to a Table Function.
+ * <p>
+ * This representation should be considered experimental. Eventually, {@link ConnectorExpression}
+ * should be extended to include this kind of argument.
+ */
+public class TableArgument
+        extends Argument
+{
+    private final Optional<QualifiedName> name;
+    private final RowType rowType;
+    private final List<String> partitionBy;
+    private final List<SortItem> orderBy;
+    private final boolean rowSemantics;
+    private final boolean pruneWhenEmpty;
+    private final boolean passThroughColumns;
+
+    @JsonCreator
+    public TableArgument(
```

Comment:
If the sub-classes of `Argument` are jackson-serialized don't we have to specify `JsonTypeInfo ` and its friends annotation for `Argument` ?